http_server/www/logout.php

```diff
@@ -5,13 +5,33 @@
 
 header("Content-type: text/plain");
 
+function is_from_game(){
+ if (isset($_SERVER["HTTP_X_REQUESTED_WITH"]) && !empty($_SERVER["HTTP_X_REQUESTED_WITH"])){
+  if (strpos($_SERVER["HTTP_X_REQUESTED_WITH"], "ShockwaveFlash/") !== 0){
+   return true;
+  }
+     else
+  {
+      return false;   
+  }
+ }
+ else
+ {
+    return false;
+ }
+}
+
 $ip = get_ip();
 
 try {
     // rate limiting
     rate_limit('logout-'.$ip, 5, 2, 'Please wait at least 5 seconds before attempting to log out again.');
     rate_limit('logout-'.$ip, 60, 10, 'Only 10 logout requests per minute per IP are accepted.');
 
+    if (is_from_game() !== true){
+     throw new Exception("For security reasons logout can only be performed from game");   
```

Comment:
Change language to keep consistent with other third-party site warnings. i.e. "It looks like you're not using PR2 to log out. For security reasons, you may only log out from a PR2 client."